Structure_Engine/Compute/NullCheck.cs

```diff
@@ -0,0 +1,186 @@
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.oM.Geometry;
+using BH.oM.Structure.Elements;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Quantities.Attributes;
+using System.ComponentModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Checks if a Node is null and outputs relevant error message.")]
+        [Input("node", "The Node to test for null.")]
+        [Input("methodName", "The name of the method to reference in the error message.")]
+        [Output("pass", "A boolean which is true if the bar passes the null check.")]
+        public static bool NullCheck(this Node node, string methodName = "Method")
+        {
+            // Check Node and Position
+            if (node?.Position == null)
+            {
+                ErrorMessage(methodName, "Node");
+                return false;
+            }
+
+            return true;
+        }
+
+        /***************************************************/
+
+        [Description("Checks if a Bar or one of its Nodes are null and outputs relevant error message.")]
+        [Input("bar", "The Bar to test for null.")]
+        [Input("methodName", "The name of the method to reference in the error message.")]
+        [Output("pass", "A boolean which is true if the bar passes the null check.")]
+        public static bool NullCheck(this Bar bar, string methodName = "Method")
+        {
+            // Check bar
+            if (bar == null)
+            {
+                ErrorMessage(methodName, "Bar");
+                return false;
+            }
+
+            // Check bar Nodes
+            return bar.StartNode.NullCheck(methodName) && bar.EndNode.NullCheck(methodName);
+        }
+
+        [Description("Checks if an FEMesh or one of its Nodes are null and outputs relevant error message.")]
+        [Input("mesh", "The FEMesh to test for null.")]
+        [Input("methodName", "Optional name of the method to reference in the error message.")]
+        [Input("checkFaces", "Optional bool to tell the method whether to check FEMeshFaces or not.")]
+        [Input("checkNodes", "Optional bool to tell the method whether to check mesh Nodes or not.")]
+        [Input("nodeListIndices", "Optional list of nodes to limit check to")]
+        [Output("pass", "A boolean which is true if the FEMeshFace passes the null check.")]
+        public static bool NullCheck(this FEMesh mesh, string methodName = "Method", bool checkFaces = true, bool checkNodes = true, List<int> nodeListIndices = default(List<int>))
```

Comment:
```suggestion
        public static bool NullCheck(this FEMesh mesh, string methodName = "Method", bool checkFaces = true, bool checkNodes = true, List<int> nodeListIndices = null)
```

Is more appropriate than `default(type)`. Then obviously handle the null in the method.

Reply:
It's been weeks, but I seem to remember trying null at first but it not working, and changing to this because of it. Will try again and see what happens.

Reply:
Feel free to gimme a shout if you need some help, happy to do a quick session over teams if needed :smile:

Reply:
Worked fine now. Not sure what happened when I originally tried it but since it seems to work now I've implemented this suggestion. Thank you for the offer, we'll see if I need it ;)